server/sonar-db-dao/src/main/java/org/sonar/db/component/BranchDao.java

```diff
@@ -37,24 +37,42 @@ public BranchDao(System2 system2) {
   }
 
   public void insert(DbSession dbSession, BranchDto dto) {
+    setKeyType(dto);
     mapper(dbSession).insert(dto, system2.now());
   }
 
   public void upsert(DbSession dbSession, BranchDto dto) {
     BranchMapper mapper = mapper(dbSession);
     long now = system2.now();
+    setKeyType(dto);
     if (mapper.update(dto, now) == 0) {
       mapper.insert(dto, now);
     }
   }
 
+  private void setKeyType(BranchDto dto) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "setKeyType" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fadd-new-columns-in-project&issues=AWGZEWsnqu6zmfejgh-9&open=AWGZEWsnqu6zmfejgh-9)

[ISSUE_KEY]: AWGZEWsnqu6zmfejgh-9